autotest/utilities/test_gdalwarp_lib.py

```diff
@@ -40,6 +40,23 @@
 
 from osgeo import gdal, ogr, osr
 
+
+@pytest.fixture(scope="module", autouse=True)
+def setup_and_cleanup():
+
+    yield
+
+    # We don't clean up when run in debug mode.
+    if gdal.GetConfigOption("CPL_DEBUG", "OFF") == "ON":
+        return
+
+    for i in range(2):
+        try:
+            os.remove("tmp/testgdalwarp" + str(i + 1) + ".tif")
+        except OSError:
+            pass
+
+
```

Comment:
I assume this isn't needed as the temporary files aren't written in tmp/ anymore

Reply:
I haven't switched to `tmp_path` everywhere, though I could follow up with that.

Reply:
fine then. can indeed be done as a follow up after this PR